pkg/sql/pg_catalog.go

```diff
@@ -1428,27 +1428,35 @@ CREATE TABLE pg_catalog.pg_type (
 `,
 	populate: func(_ context.Context, p *planner, addRow func(...parser.Datum) error) error {
 		h := makeOidHasher()
-		for oid, typ := range parser.OidToType {
+		for o, typ := range parser.OidToType {
 			cat := typCategory(typ)
 			typElem := oidZero
 			builtinPrefix := parser.PGIOBuiltinPrefix(typ)
 			if cat == typCategoryArray {
-				typElem = parser.NewDOid(parser.DInt(parser.UnwrapType(typ).(parser.TArray).Typ.Oid()))
-				if typ != parser.TypeIntVector {
+				if typ == parser.TypeIntVector {
+					// IntVector needs a special case because its a special snowflake
+					// type. It's just like an Int2Array, but it has its own OID. We
+					// can't just wrap our Int2Array type in an OID wrapper, though,
+					// because Int2Array is not an exported, first-class type - it's an
+					// input-only type that translates immediately to int8array. This
+					// would go away if we decided to export Int2Array as a real type.
```

Comment:
So are there any downsides to exporting `typeInt2Array`?

Also, I'm not sure I get why doing `TypeIntVector = wrapTypeWithOid(typeInt2Array, oid.T_int2vector)` without exporting `typeInt2Array` wouldn't work. Seems like the only other changes would be adding `case parser.TypeIntVector:` as a case to `sql.checkResultType` and may changing something around `NewDIntVectorFromDArray`. I might be missing something though.